In [27]:
import pandas as pd
import numpy as np 
import matplotlib.pyplot as plt 
import pickle 
from pandas_profiling import ProfileReport
import warnings
warnings.filterwarnings("ignore")

In [2]:
df=pd.read_csv("advertising.csv")

In [3]:
df.head()

TV  Radio  Newspaper  Sales
0  230.1   37.8       69.2   22.1
1   44.5   39.3       45.1   10.4
2   17.2   45.9       69.3   12.0
3  151.5   41.3       58.5   16.5
4  180.8   10.8       58.4   17.9

In [4]:
df.shape

(200, 4)

In [5]:
df.size

800

In [6]:
df.isnull().sum()

TV           0
Radio        0
Newspaper    0
Sales        0
dtype: int64

In [7]:
df.describe()

TV       Radio   Newspaper       Sales
count  200.000000  200.000000  200.000000  200.000000
mean   147.042500   23.264000   30.554000   15.130500
std     85.854236   14.846809   21.778621    5.283892
min      0.700000    0.000000    0.300000    1.600000
25%     74.375000    9.975000   12.750000   11.000000
50%    149.750000   22.900000   25.750000   16.000000
75%    218.825000   36.525000   45.100000   19.050000
max    296.400000   49.600000  114.000000   27.000000

In [8]:
pr=ProfileReport(df)

In [9]:
pr.to_widgets()

Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render widgets:   0%|          | 0/1 [00:00<?, ?it/s]

In [10]:
pr.to_file("Advertising Test Dashboard.html")

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

In [11]:
X=df.drop("Sales",axis=1)

In [12]:
y=df["Sales"]

In [13]:
from sklearn.linear_model import LinearRegression

In [14]:
model= LinearRegression()

In [15]:
model.fit(X,y)  #we will be able to find the value of "m" and "c".

LinearRegression()

In [16]:
model.intercept_  #intercept(y-intercept) is the value of c. 

4.625124078808655

In [17]:
model.coef_  #value of "m" for three indenpendent variable TV, Newspaper, Radio.

array([0.05444578, 0.10700123, 0.00033566])

#we know that the equation of a straight line is, y=mx+c

Andd every thing, like all the mathematical calcualtions that need to be done, to find out the best value of m and c is done into a single line which is model.fit(X,y`)

we know that, sales= m*TV + c

thats how all the calculation and training is done within fit. Training is nothing but finding the best equation of the model.

In [37]:
model.predict([[30,40,45]])   #y=mx+c   where the parameters are the value of x.

array([10.55365122])

In this case, the predicted output is approximately 10.55365122. This means that, according to your trained linear regression model, when the values of the three features are 30, 40, and 45, it predicts the target variable (dependent variable) to be approximately 10.55365122

In [38]:
file="linear_reg.sav"
pickle.dump(model,open(file,'wb'))

In [39]:
saved_model= pickle.load(open(file,"rb"))

In [40]:
saved_model.predict([[30,40,50]])

array([10.55532951])

In [41]:
model.score(X,y)

0.9025912899684558